lib/src/lints/avoid_using_debug_print/avoid_using_debug_print_rule.dart

```diff
@@ -0,0 +1,128 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/syntactic_entity.dart';
+import 'package:analyzer/error/listener.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:solid_lints/src/lints/avoid_using_debug_print/models/avoid_using_debug_print_func_model.dart';
+import 'package:solid_lints/src/models/rule_config.dart';
+import 'package:solid_lints/src/models/solid_lint_rule.dart';
+
+/// A `avoid_using_debug_print` rule which forbids calling or referencing
+/// debugPrint function from flutter/foundation.
+///
+/// ### Example
+/// 
+/// #### BAD:
+///
+/// ```dart
+/// debugPrint(''); // LINT
+/// var ref = debugPrint; // LINT
+/// var ref2;
+/// ref2 = debugPrint; // LINT
+/// ```
+///
+/// #### GOOD:
+///
+/// ```dart
+/// log('');
+/// ```
+class AvoidUsingDebugPrint extends SolidLintRule {
+  /// The [LintCode] of this lint rule that represents
+  /// the error when debugPrint is called
+  static const lintName = 'avoid_using_debug_print';
+
+  AvoidUsingDebugPrint._(super.config);
+
+  /// Creates a new instance of [AvoidUsingDebugPrint]
+  /// based on the lint configuration.
+  factory AvoidUsingDebugPrint.createRule(CustomLintConfigs configs) {
+    final rule = RuleConfig(
+      configs: configs,
+      name: lintName,
+      problemMessage: (_) => "Avoid using 'debugPrint'",
+    );
+
+    return AvoidUsingDebugPrint._(rule);
+  }
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ErrorReporter reporter,
+    CustomLintContext context,
+  ) {
+    context.registry.addFunctionExpressionInvocation(
+      (node) {
+        final func = node.function;
+        if (func is! Identifier) {
+          return;
+        }
+        _checkIdentifier(
+          identifier: func,
+          node: node,
+          reporter: reporter,
+        );
+      },
+    );
+
+    context.registry.addVariableDeclaration((node) {
```

Comment:
Perhaps worth leaving a note here that `FunctionReference` doesn't work and we're using VariableDeclaration as a workaround.
```suggestion
    // addFunctionReference does not get triggered...
    context.registry.addVariableDeclaration((node) {
```